opencti-platform/opencti-front/src/private/components/settings/fintel_design/FintelDesignForm.tsx

```diff
@@ -0,0 +1,135 @@
+import { FintelDesign_fintelDesign$data } from '@components/settings/fintel_design/__generated__/FintelDesign_fintelDesign.graphql';
+import { graphql } from 'react-relay';
+import CustomFileUploader from '@components/common/files/CustomFileUploader';
+import { FormikConfig } from 'formik/dist/types';
+import { fieldSpacingContainerStyle } from '../../../../utils/field';
+import ColorPickerField from '../../../../components/ColorPickerField';
+import { FintelDesignFormValues } from './FintelDesign';
+import { useFormatter } from '../../../../components/i18n';
+import useApiMutation from '../../../../utils/hooks/useApiMutation';
+import { handleErrorInForm } from '../../../../relay/environment';
+
+const fintelDesignFormFieldPatchMutation = graphql`
+  mutation FintelDesignFormFieldPatchMutation($id: ID!, $input: [EditInput!], $file: Upload) {
+    fintelDesignFieldPatch(id: $id, input: $input, file: $file) {
+      id
+      name
+      file_id
+      description
+      gradiantFromColor
+      gradiantToColor
+      textColor
+    }
+  }
+`;
+
+interface FintelDesignFormProps {
+  onChange: (val: FintelDesignFormValues) => void;
+  fintelDesign: FintelDesign_fintelDesign$data;
+}
+
+const FintelDesignForm: FunctionComponent<FintelDesignFormProps> = ({ onChange, fintelDesign }) => {
+  const { t_i18n } = useFormatter();
+
+  const initialValues: FintelDesignFormValues = {
+    file: null,
+    gradiantFromColor: fintelDesign.gradiantFromColor,
+    gradiantToColor: fintelDesign.gradiantToColor,
+    textColor: fintelDesign.textColor,
+  };
+
+  const fintelDesignValidation = () => Yup.object().shape({
+    gradiantFromColor: Yup.string().nullable(),
+    gradiantToColor: Yup.string().nullable(),
+    textColor: Yup.string().nullable(),
+  });
+
+  const [commitFieldPatch] = useApiMutation(fintelDesignFormFieldPatchMutation);
+
+  const onSubmit: FormikConfig<FintelDesignFormValues>['onSubmit'] = (
+    values,
+    { setSubmitting, setErrors },
+  ) => {
+    setSubmitting(true);
+    const { file } = values;
+    const inputValues = Object.entries(values)
+      .filter(([key, _]) => !['file'].includes(key))
+      .map(([key, value]) => ({ key, value: value ?? [] }));
+
+    commitFieldPatch({
+      variables: {
+        id: fintelDesign.id,
+        input: inputValues,
+        file,
+      },
+      onCompleted: () => {
+        setSubmitting(false);
+        onChange(values);
+      },
+      onError: (error: Error) => {
+        handleErrorInForm(error, setErrors);
+        setSubmitting(false);
+      },
+    });
+  };
+
+  return (
+    <Formik<FintelDesignFormValues>
+      onSubmit={onSubmit}
+      enableReinitialize={true}
+      initialValues={initialValues}
+      validationSchema={fintelDesignValidation}
+      validateOnChange={true}
+      validateOnBlur={true}
+    >
+      {({ setFieldValue, submitForm }) => {
+        return (
+          <Form>
+            <Field
+              component={CustomFileUploader}
```

Comment:
I suggest some improvements for this field:

```suggestion
              component={CustomFileUploader}
              label={t_i18n('Logo')}
              acceptMimeTypes="image/jpeg,image/png,image/svg+xml"
```
`label`: default is "associated file", "logo" seems clearer IMO.
`acceptMimeTypes`: I have tested with a `.webp` file, and it crashed the rendering. I think for safety we can accept only these images types.

Reply:
Follow-up on this: I tried to encapsulate between try/catch clauses the code that do the rendering of the PDF preview, but I keep having exception thrown outside of Promises "invalid image format" or something.

@lndrtrbn @SarahBocognano  if you have any idea.... I simply wanted to do a `$MESSAGING.notifiyError(...)` when something goes bad. Right now (you can test by using a webp image) the preview paper is simply empty without any visible error.

Reply:
With the `acceptedMimeType` it lower the chance of having an issue, but still. In the end, user is still capabale of using a bad image file.

Reply:
Ok I'll have a look